lib/quick_travel/adapter.rb

```diff
@@ -104,19 +104,17 @@ def self.check_id!(id)
     def self.find_all!(request_path, opts = {})
       response = if opts.key? :cache
         QuickTravel::Cache.cache(opts[:cache], opts[:cache_options]) {
-          get_and_validate(request_path, opts.except(:cache, :cache_options))
+          get_and_validate(request_path, opts.except(:cache, :cache_options), return_response_object: true)
```

Comment:
Still ultimately have the same problem, get_and_validate is still returning multiple different return values determined by an input parameter.
if get_and_validate cannot be changed to use the response object everywhere than create a simple wrapper method around get_and_validate which converts the response to cacheable.

i.e.
```
build_cacheable_response( get_and_validate(......) )
```